Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponentMode.cpp

```diff
@@ -143,6 +144,33 @@ namespace GradientSignal
         ImageGradientModificationNotificationBus::Handler::BusDisconnect();
     }
 
+    void EditorImageGradientComponentMode::Reflect(AZ::ReflectContext* context)
+    {
+        AzToolsFramework::ComponentModeFramework::ReflectEditorBaseComponentModeDescendant<EditorImageGradientComponentMode>(context);
+    }
+
+    void EditorImageGradientComponentMode::RegisterActions()
+    {
+    }
+
+    void EditorImageGradientComponentMode::BindActionsToModes()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "EditorImageGradientComponentMode - could not get ActionManagerInterface on BindActionsToModes.");
+
+        AZ::SerializeContext* serializeContext = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationRequests::GetSerializeContext);
+
+        AZStd::string modeIdentifier = AZStd::string::format(
+            "o3de.context.mode.%s", serializeContext->FindClassData(azrtti_typeid<EditorImageGradientComponentMode>())->m_name);
+
```

Comment:
Suggestion: Consider moving all this boilerplate down into PaintBrushManipulator::BindActionsToMode() and just pass down `azrtti_typeid<EditorImageGradientComponentMode>()` as the method input.